cscs-checks/system/slurm/slurm.py

```diff
@@ -285,9 +286,12 @@ class slurm_response_check(rfm.RunOnlyRegressionTest):
     def set_exec_opts(self):
         self.executable_opts = [self.command]
 
-    @run_before('sanity')
-    def set_sanity(self):
-        self.sanity_patterns = sn.assert_eq(self.job.exitcode, 0)
+    @sanity_function
+    def assert_exitcode_zero(self):
+        return sn.assert_eq(self.job.exitcode, 0)
+
+    @run_before('performance')
+    def set_perf_patterns(self):
         self.perf_patterns = {
             'real_time': sn.extractsingle(r'real (?P<real_time>\S+)',
                                           self.stderr, 'real_time', float)
```

Comment:
```suggestion
    @performance_function('s')
    def real_time(self):
        return sn.extractsingle(r'real (?P<real_time>\S+)', self.stderr,
                                'real_time', float)
```

Reply:
Hmm, I think the reference of this test is not being used. Unless, I'm missing something, I think that one would have to transform that `reference` in to the right reference for each value of the `command` parameter.
```python
self.reference = {
    '*': self.reference[self.command],
}
```